### ISSUE TITLE
Support typing.ForwardRef declarations

### ISSUE THREAD

**Issue by koliyo** [PyCQA/pylint#2540]
### Current behavior

With this code
```py
from typing import ForwardRef

Foo = ForwardRef('Foo')

class Foo:
    pass
```

I get 
```
foo.py:5:0: E0102: class already defined line 3 (function-redefined)
```

### Expected behavior
I would expect pylint to support this construction

### pylint --version output

```
$ python3 -m pylint --version
__main__.py 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jul 23 2018, 20:22:55)
[Clang 9.1.0 (clang-902.0.39.2)]
```

**Comment by PCManticore** [comment]
Thanks for creating an issue! This definitely makes sense.
